core/trino-main/src/main/java/io/trino/server/remotetask/HttpRemoteTask.java

```diff
@@ -596,6 +629,36 @@ private void triggerUpdate()
         }
     }
 
+    /**
+     * Adaptively adjust batch size to meet expected request size:
+     * If requestSize is not equal to expectedSize, this function will try to estimate and adjust the batch size proportionally based on
+     * current nums of splits and size of request.
+     */
+    private int adjustSplitBatchSize(List<SplitAssignment> splitAssignments, long requestSize, long expectedSize, int currentSplitBatchSize)
```

Comment:
I might be interpreting this method incorrectly. This method assumes that the `requestSize` is the size of a request that includes all available splits. However since there's a limit on the number of splits the current request size might only include the number of splits equal to `currentSplitBatchSize`.

Reply:
Oh, right. Nevermind. I must've misread.